tests/src/Interop/COM/NETClients/Primitives/ArrayTests.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace ComClient
+{
+    using CoreFXTestLibrary;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    class ArrayTests
+    {
+        static private readonly IEnumerable<int> BaseData = Enumerable.Range(0, 10);
+
+        private readonly Server.Contract.IArrayTesting server;
+        private readonly double expectedMean;
+
+        public ArrayTests()
+        {
+            this.server = (Server.Contract.IArrayTesting)new Server.Contract.Servers.ArrayTestingClass();
+
+            double acc = 0.0;
+            int[] rawData = BaseData.ToArray();
+            foreach (var d in rawData)
+            {
+                acc += d;
+            }
+
+            expectedMean = acc / rawData.Length;
+        }
+
+        public void Run()
+        {
+            this.Marshal_ByteArray();
+            this.Marshal_ShortArray();
+            this.Marshal_UShortArray();
+            this.Marshal_IntArray();
+            this.Marshal_UIntArray();
+            this.Marshal_LongArray();
+            this.Marshal_ULongArray();
+            this.Marshal_FloatArray();
+            this.Marshal_DoubleArray();
+        }
+
+        static private bool EqualByBound(double expected, double actual)
+        {
+            double low = expected - 0.00001;
+            double high = expected + 0.00001;
+            double eps = Math.Abs(expected - actual);
+            bool isEqual = eps < double.Epsilon || (low < actual && actual < high);
+            if (!isEqual)
+            {
+                Console.WriteLine($"{expected}  {actual}");
+            }
+
+            return isEqual;
+        }
+
+        private void Marshal_ByteArray()
+        {
+            int len;
+            byte[] data = BaseData.Select(i => (byte)i).ToArray();
+
+            Console.WriteLine($"{data.GetType().Name} marshalling");
+            Assert.IsTrue(EqualByBound(expectedMean, this.server.Mean_Byte_LP_PreLen(data.Length, data)), $"Mean_Byte_LP_PreLen");
+            Assert.IsTrue(EqualByBound(expectedMean, this.server.Mean_Byte_LP_PostLen(data, data.Length)), $"Mean_Byte_LP_PostLen");
+            Assert.IsTrue(EqualByBound(expectedMean, this.server.Mean_Byte_SafeArray_OutLen(data, out len)), $"Mean_Byte_SafeArray_OutLen");
```

Comment:
Suggestion: Could you split LPArray and SafeArray into two files(two clients)?

Reply:
Is there a specific reason for that? I wrote this test to be as basic as possible. In the sense that these tests represent a minimum stable platform.